db/deploy/mutations/archive_application_sow.sql

```diff
@@ -14,7 +14,7 @@ begin
     sow_data_ids := array(
         select sd.id
         from ccbc_public.application_sow_data as sd
-        where sd.amendment_number = _amendment_number
+        where sd.amendment_number = _amendment_number and sd.application_id = _application_id
     );
 
     -- Update archived_at for non-archived records in application_sow_data table
```

Comment:
I think we need the same check here as well?

Reply:
Ah yes you're right, let me update that. Thanks!